packages/wallets/phantom/src/index.ts

```diff
@@ -3,42 +3,142 @@ import {
   type ConnectWalletParams,
   SwapKitError,
   WalletOption,
+  ensureEVMApiKeys,
   setRequestClientConfig,
 } from "@swapkit/helpers";
+import type { SolanaProvider } from "@swapkit/toolbox-solana";
 
-function getPhantomProvider() {
-  // @ts-ignore
-  return window.phantom?.solana;
+export const PHANTOM_SUPPORTED_CHAINS = [Chain.Bitcoin, Chain.Ethereum, Chain.Solana] as const;
+export type PhantomSupportedChains = (typeof PHANTOM_SUPPORTED_CHAINS)[number];
+
+declare global {
+  interface Window {
+    phantom: {
+      solana: SolanaProvider;
+    };
+  }
 }
 
-function connectPhantom({ addChain, config: { thorswapApiKey }, rpcUrls }: ConnectWalletParams) {
-  return async function connectPhantom(chain: Chain = Chain.Solana) {
-    setRequestClientConfig({ apiKey: thorswapApiKey });
+async function getPhantomProvider<T extends PhantomSupportedChains>(chain: T) {
```

Comment:
@chillios-dev , we are using `switch case` in both `getPhantomProvider` and `getWalletMethods` for the same cases.
Embedding `getPhantomProvider` into `getWalletMethods` could be good imo. What is your opinion?

CC: @towanTG

Reply:
Makes sense - I rushed split too soon it seems 😅